src/snowflake/connector/cursor.py

```diff
@@ -704,6 +710,11 @@ def execute_async(self, *args, **kwargs):
         kwargs["_exec_async"] = True
         return self.execute(*args, **kwargs)
 
+    def describe(self, *args, **kwargs):
+        """Obtain the schema/description of the result without executing the query"""
+        kwargs["_describe_only"] = kwargs["_is_internal"] = True
+        return self.execute(*args, **kwargs)
```

Comment:
Should we not return the schema explicitly instead of an empty resultset?

Reply:
Yes, let's return the schema directly.